### PR TITLE
Typescriptify Frontend

### PR DESCRIPTION
**Pull request by Simon-Laux**
- [X] LoginScreen.js
- [X] contact list
- [X] message
- [X] Menu.js

note: rebase mute notification branch after merging this pr (because of Menu.js)

### CODE REVIEW

File: src/renderer/components/helpers/ChatMethods.ts [hunk 1]
```diff
@@ -94,10 +95,9 @@ export function openMapDialog(
 
 export async function openViewProfileDialog(
   screenContext: unwrapContext<typeof ScreenContext>,
-  contact: number | JsonContact
+  contact_id: number
```
Comment (Jikstra) [inline src/renderer/components/helpers/ChatMethods.ts:98]: Are you sure that we never call it with just a contact?

File: src/renderer/components/message/Message.tsx [hunk 2]
```diff
@@ -26,7 +33,7 @@ const Avatar = (contact, onContactClick) => {
       ? String.fromCodePoint(codepoint).toUpperCase()
       : '#'
     return (
-      <div className='author-avatar default' alt={alt} onClick={onClick}>
+      <div className='author-avatar default' aria-label={alt} onClick={onClick}>
```
Comment (Jikstra) [inline src/renderer/components/message/Message.tsx:36]: Shouldn't we have alt + aria-label? I guess the idea of the alt war that you can hover over it and see the name. Maybe not needed here ux wise?

File: src/renderer/components/message/MessageList.tsx [hunk 3]
```diff
@@ -33,7 +45,7 @@ export default function MessageList({
     chatStoreDispatch,
   ] = useChatStore()
   const messageListRef = useRef(null)
-  const lastKnownScrollHeight = useRef([null, null])
+  const lastKnownScrollHeight = useRef<number>(null)
```
Comment (Jikstra) [inline src/renderer/components/message/MessageList.tsx:48]: Isn't this changing it from an array with two null elements to a single null ?

File: src/renderer/components/message/MessageWrapper.tsx [hunk 4]
```diff
@@ -18,30 +28,28 @@ export const render = props => {
...
     const tx = window.translate
     const screenContext = useContext(ScreenContext)
     const { openDialog } = screenContext
 
-    message.onForward = () => openDialog('ForwardMessage', { message })
-
-    const conversationType = GROUP_TYPES.includes(chat.type)
+    const conversationType: 'group' | 'direct' = GROUP_TYPES.includes(chat.type)
       ? 'group'
       : 'direct'
```
Comment (Jikstra) [inline src/renderer/components/message/MessageWrapper.tsx:41]: This is one of the things i dont like about typescript, you have the types there twice and it's getting really verbose. I sometimes have the feeling that I need to duplicate type annotations quite a lot.

File: src/renderer/components/helpers/ChatMethods.ts (hunk 1 above)
Comment (Simon-Laux) [reply src/renderer/components/helpers/ChatMethods.ts:98]: yes I changed it everywhere

File: src/renderer/components/message/Message.tsx (hunk 2 above)
Comment (Simon-Laux) [reply src/renderer/components/message/Message.tsx:36]: alt is not a valid property on div (in react), so we should probably remove the thing here.

File: src/renderer/components/message/MessageList.tsx (hunk 3 above)
Comment (Simon-Laux) [reply src/renderer/components/message/MessageList.tsx:48]: I'm not sure what the array should represent here other than an typescript error.

File: src/renderer/components/message/MessageWrapper.tsx (hunk 4 above)
Comment (Simon-Laux) [reply src/renderer/components/message/MessageWrapper.tsx:41]: the typescript compile is not the smartest... But if it were, build times would probably be longer.

File: src/renderer/components/message/MessageList.tsx (hunk 3 above)
Comment (Jikstra) [reply src/renderer/components/message/MessageList.tsx:48]: Yup we initialized it as an array but set it to a single number ^^ Typical js bug that's not possible with typescript.

**Pull request merged**